src/ingester/mod.rs

```diff
@@ -1,22 +1,71 @@
+
 use std::thread::sleep;
 use std::time::Duration;
 
 use error::IngesterError;
 use futures::stream::StreamExt;
 use futures::TryStreamExt;
 use parser::parse_transaction;
+use sea_orm::sea_query::OnConflict;
+use sea_orm::ConnectionTrait;
 use sea_orm::DatabaseConnection;
 use sea_orm::DatabaseTransaction;
+
+use sea_orm::EntityTrait;
+use sea_orm::QueryTrait;
+use sea_orm::Set;
 use sea_orm::TransactionTrait;
+
 use typedefs::block_info::TransactionInfo;
 
 use self::typedefs::block_info::BlockInfo;
+use crate::dao::generated::blocks;
 pub mod error;
 pub mod fetchers;
 pub mod parser;
 pub mod persist;
 pub mod typedefs;
 
+pub async fn index_block(db: &DatabaseConnection, block: &BlockInfo) -> Result<(), IngesterError> {
+    let txn = db.begin().await?;
+    index_block_metadata(&db, block).await?;
+    for transaction in &block.transactions {
+        index_transaction(&txn, &transaction, block.slot).await?;
+    }
+    txn.commit().await?;
+
+    Ok(())
+}
+
+async fn index_block_metadata(
```

Comment:
is this the new audit log?

Reply:
Yes this used for auditing but I am not sure that it will be a full replacement of the existing audit log. Might add more auditing data later.